### PR TITLE
DTLS CID Support

### PR DESCRIPTION
**Pull request by plskeggs**
This replaces https://github.com/zephyrproject-rtos/zephyr/pull/44914.

This updates commits from https://github.com/zephyrproject-rtos/zephyr/pull/36738, and builds on them.

- update for latest main branch
- add function to query the peer's CID

### CODE REVIEW

File: include/zephyr/net/socket.h [hunk 1]
```diff
@@ -198,6 +214,12 @@ struct zsock_addrinfo {
 	char _ai_canonname[DNS_MAX_NAME_SIZE + 1];
 };
 
+struct tls_dtls_peer_cid {
+	int enabled;
+	unsigned char peer_cid[32];
```
Comment (ceolin) [inline include/zephyr/net/socket.h:219]: s/32/MBEDTLS_SSL_CID_OUT_LEN_MAX/g ?

**Review by ceolin — DISMISSED** [review]

**Review by hasheddan — APPROVED** [review]
Thanks @plskeggs!

**Comment by plskeggs** [comment]
I want to make more changes, so in order prevent merging yet, I am moving this to draft status.

File: subsys/net/lib/sockets/sockets_tls.c [hunk 2]
```diff
@@ -1613,6 +1651,152 @@ static int tls_opt_dtls_handshake_timeout_set(struct tls_context *context,
...
+		return -EINVAL;
+	}
+
+	int value = *((int *)optval);
+
+	switch (value) {
+	case TLS_DTLS_CID_DISABLED:
+		context->options.dtls_cid.enabled = false;
+		context->options.dtls_cid.cid_len = 0;
+	break;
```
Comment (rlubos) [inline subsys/net/lib/sockets/sockets_tls.c:1673]: Wrong indent.

File: subsys/net/lib/sockets/sockets_tls.c [hunk 3]
```diff
@@ -1613,6 +1651,152 @@ static int tls_opt_dtls_handshake_timeout_set(struct tls_context *context,
...
+	case TLS_DTLS_CID_SUPPORTED:
+		context->options.dtls_cid.enabled = true;
+		context->options.dtls_cid.cid_len = 0;
+	break;
```
Comment (rlubos) [inline subsys/net/lib/sockets/sockets_tls.c:1677]: Wrong indent.

**Review by rlubos — COMMENTED** [review]
Looks mostly good, however I'd squash `net: sockets: tls: Implement TLS_DTLS_CONNECTION_ID option ` and `net: sockets: tls: add ability to get peer CID` commits to make history cleaner, no need to introduce intermediate solutions just to have it modified in the next commit.

**Review by rlubos — DISMISSED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Review by rlubos — APPROVED** [review]

**Pull request merged**